### PR TITLE
[Compiler] Activate integration tests & Update codecov and docstrings

### PR DESCRIPTION
**Pull request by maliasadi**
This PR updates the support and addresses code review suggestions in PR #4692.

- [x] Activate integration tests in CI (using Catalyst v0.3.2)
- [x] Fix docstring issues in the compiler and grad docs
- [x] Use `h` instead of `step_size` in grad ops
- [x] Update code coverage of the compiler module

### CODE REVIEW

File: doc/introduction/compiling_circuits.rst
Comment (maliasadi) [inline doc/introduction/compiling_circuits.rst:445]: @josh146 what are your thoughts on temporarily removing this until we establish a stable API for it?

File: pennylane/_grad.py
Comment (maliasadi) [inline pennylane/_grad.py:30]: @josh146 I removed the qjit support from the headline. We link to `catalyst.grad` and `catalyst.jacobian` doc pages for more information in these doc pages as well as there's now a new list of QJIT compatible PL ops in the compiler doc page. I can revert this if you think otherwise. cc:  @timmysilv

File: pennylane/compiler/__init__.py
Comment (maliasadi) [inline pennylane/compiler/__init__.py:67]: @josh146 what are your opinions on the addition of this list?

File: pennylane/compiler/qjit_api.py
Comment (maliasadi) [inline pennylane/compiler/qjit_api.py:108]: @josh146 @timmysilv with adding "no-cover" only in 2-places for skipping such cases and updating the integration tests, I could get %100 codecov. lmk if this is not a common practice in PL and I can get rid of these `pragma`s.

Screenshot from 2023-11-10 15-37-17

File: pennylane/compiler/__init__.py
Comment (timmysilv) [inline pennylane/compiler/__init__.py:182]: i would make it extra-obvious, just in case. like in its own paragraph before, put something like:

> 🚨 The PennyLane compiler API is experimental and subject to change 🚨

File: pennylane/compiler/qjit_api.py
Comment (timmysilv) [reply pennylane/compiler/qjit_api.py:108]: I think it's fine for errors like these. otherwise you could mock the `compiler` module with monkeypatch, but it's not the biggest deal

File: pennylane/_grad.py
Comment (timmysilv) [inline pennylane/_grad.py:66]: I'm a little torn about removing these from docstrings, but I'll leave it up to you/other reviewers

**Review by timmysilv — COMMENTED** [review]
these changes look great, ty!

File: doc/introduction/compiling_circuits.rst
Comment (josh146) [reply doc/introduction/compiling_circuits.rst:445]: I would like to leave this in :)

File: pennylane/_grad.py
Comment (josh146) [reply pennylane/_grad.py:66]: Yep, how come this is being removed? Please leave this in, as we should be documenting all keyword arguments of a function!

File: pennylane/_grad.py
Comment (josh146) [inline pennylane/_grad.py:237]: Same here, can we please keep this in :)

File: pennylane/compiler/__init__.py
Comment (josh146) [reply pennylane/compiler/__init__.py:67]: I like it! Good idea to include this.

File: pennylane/compiler/__init__.py
Comment (josh146) [reply pennylane/compiler/__init__.py:182]: Let's make this a warning directive.

File: pennylane/compiler/__init__.py
Comment (josh146) [inline pennylane/compiler/__init__.py:185]:
```suggestion
.. warning::

    The PennyLane compiler API is experimental and subject to change.

To register any compiler packages, an experimental interface is available.
This interface exposes the ``entry_points``
metadata under the designated group name ``pennylane.compilers``, including the
following entry points:
```

File: pennylane/compiler/compiler.py
Comment (josh146) [inline pennylane/compiler/compiler.py:43]: Nice code comment!

File: pennylane/compiler/compiler.py
Comment (josh146) [inline pennylane/compiler/compiler.py:121]: good catch

**Review by josh146 — CHANGES_REQUESTED** [review]
@maliasadi have left some suggestions. Overall looking good, but the Args: for `jacobian` and `grad` should be re-added before this can be merged.

File: pennylane/_grad.py
Comment (timmysilv) [inline pennylane/_grad.py:81]:
```suggestion
        h (float): The step-size value for the finite-difference (``"fd"``) method within
```

File: pennylane/_grad.py
Comment (timmysilv) [inline pennylane/_grad.py:71]: the indentation below here doesn't seem to be rendering correctly

File: pennylane/_grad.py
Comment (timmysilv) [inline pennylane/_grad.py:225]: same concern

File: pennylane/_grad.py
Comment (timmysilv) [inline pennylane/_grad.py:235]:
```suggestion
        h (float): The step-size value for the finite-difference (``"fd"``) method within
```

**Review by timmysilv — APPROVED** [review]
just some little things, otherwise lgtm!

File: pennylane/_grad.py
Comment (maliasadi) [reply pennylane/_grad.py:71]: I'm still trying to fix the issue with indentation  :( it's not the final version

Comment (maliasadi) [reply pennylane/_grad.py]: I couldn't fix the issue with the new line in docs :( Any ideas?

Screenshot from 2023-11-14 15-27-18

Comment (timmysilv) [reply pennylane/_grad.py]: not sure. imho this is not terrible, and I'm ok with it as it is

**Review by josh146 — APPROVED** [review]
Thanks @maliasadi!

**Pull request merged**
